### PR TITLE
docs: add docs on global config

### LINKED ISSUE #305: docs: Document `config` usage
Please document what is available by using `config`.

We spent quite some time to figure out how to spare us the work of setting up each `mount`ing individually and configure the `global.mocks` **globally**.

So instead of:
```javascript
it('test', () => {
  const component1 = mount(Component1, {global: {mocks: {$t: message => message}}});
  const component2 = mount(Component2, {global: {mocks: {$t: message => message}}});
})
```

You can use 
```javascript
import {config} from 'vue-test-utils';

config.global.mocks = {$t: message => message};

it('test', () => {
  const component1 = mount(Component1);
  const component2 = mount(Component2);
})
```

### PR DESCRIPTION
**Pull request by lmiller1990**
resolves #305

We need to finish docs soon, it's just a few hours of work. Let's do it.

### CODE REVIEW

File: docs/api/index.md [hunk 1]
```diff
@@ -1274,3 +1270,44 @@ test('unmount', () => {
...
+  },
+  template: `
+    <p>{{ $t('message') }}</p>
+    <my-component />
+  `
+}
+
+it('uses global config', () => {
+  const wrapper = mount(Component)
+  console.log(wrapper.html()) // <p>message</p><div />
```
Comment (cexbrayat) [inline docs/api/index.md:1311]: add the opening `<div>` or remove the closing one

**Review by cexbrayat — APPROVED** [review]

**Comment by cexbrayat** [comment]
Unrelated to the PR, but it would be nice to extract the examples into real code (that's what I do in my ebooks, and that's life saving for maintainability). Has this been looked into?

**Comment by afontcu** [comment]
> Unrelated to the PR, but it would be nice to extract the examples into real code (that's what I do in my ebooks, and that's life saving for maintainability). Has this been looked into?

Not… yet 😉  https://github.com/vuejs/vue-test-utils-next/issues/301

**Comment by afontcu** [comment]
looks good!


we do have a list of `global` options above the point you introduce here:

imatge

Could we link them somehow?

**Comment by lmiller1990** [comment]
@cexbrayat sure, fixed

@afontcu what do you think of this tip:

image

**Review by afontcu — APPROVED** [review]
Great 🚀

**Pull request merged**